packages/vite/src/node/server/middlewares/static.ts

```diff
@@ -175,7 +175,7 @@ export function serveRawFsMiddleware(
       let newPathname = pathname.slice(FS_PREFIX.length)
       if (isWindows) newPathname = newPathname.replace(/^[A-Z]:/i, '')
 
-      url.pathname = encodeURIComponent(newPathname)
+      url.pathname = encodeURI(newPathname)
```

Comment:
This part was returning `%2Ffoo%2Fbar` when `url.pathname` was `/foo/bar`. But because https://github.com/lukeed/sirv/pull/149 changed `decodeURIComponent` to `decodeURI`, this no longer works.

The reason why this part was using `encodeURIComponent` is #8979. In that time, we weren't able to use `decodeURI` in line 162 because of #8804.